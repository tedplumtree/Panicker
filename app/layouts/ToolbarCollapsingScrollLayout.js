import React, { PropTypes } from 'react';
import {
  ScrollView,
  StatusBar,
  StyleSheet,
  View,
} from 'react-native';
import { Toolbar } from 'react-native-material-ui';

const propTypes = {
  children: PropTypes.any.isRequired,
  toolbar: PropTypes.object.isRequired,
};

const contextTypes = {
  uiTheme: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

const ToolbarCollapsingScrollLayout = ({ children, toolbar }, { uiTheme }) => (
  <View style={styles.container}>
    <StatusBar backgroundColor={uiTheme.palette.primaryColor} />
    <ScrollView style={styles.container}>
      <Toolbar {...toolbar} />
      {children}
    </ScrollView>
  </View>
);

ToolbarCollapsingScrollLayout.propTypes = propTypes;
ToolbarCollapsingScrollLayout.contextTypes = contextTypes;

export default ToolbarCollapsingScrollLayout;
