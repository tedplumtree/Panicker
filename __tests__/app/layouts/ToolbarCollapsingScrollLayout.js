import { Text } from 'react-native';
import { ThemeProvider } from 'react-native-material-ui';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ToolbarCollapsingScrollLayout from '../../../app/layouts/ToolbarCollapsingScrollLayout';

const dummyMaterialDesignTheme = {
  uiTheme: {
    palette: {
      primaryColor: '#009688',
    },
    toolbar: {
      container: {
        height: 123,
      },
    },
  },
};

const dummyToolbarProps = {
  leftElement: 'menu',
  centerElement: 'Foo bar test center element',
  searchable: {
    autoFocus: true,
    placeholder: 'Search this thing!',
  },
};

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider uiTheme={dummyMaterialDesignTheme}>
      <ToolbarCollapsingScrollLayout toolbar={dummyToolbarProps}>
        <Text>Some dummy child component</Text>
        <Text>Another dummy child component</Text>
      </ToolbarCollapsingScrollLayout>
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
