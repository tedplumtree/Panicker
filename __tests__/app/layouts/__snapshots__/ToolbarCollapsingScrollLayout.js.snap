// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2196f3",
            "flexDirection": "row",
            "height": 56,
            "overflow": "hidden",
            "paddingHorizontal": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#2196f3",
                "borderRadius": 0,
                "height": 0,
                "position": "absolute",
                "right": undefined,
                "top": -0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 0,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 0,
                "height": 0,
                "left": undefined,
                "position": "absolute",
                "top": -0,
                "transform": Array [
                  Object {
                    "scale": 0.01,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={undefined}
            testID={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderRadius": 21.6,
                  "height": 43.2,
                  "left": 2.3999999999999986,
                  "opacity": 0.16,
                  "position": "absolute",
                  "top": 2.3999999999999986,
                  "transform": Array [
                    Object {
                      "scale": 0.01,
                    },
                  ],
                  "width": 43.2,
                  "zIndex": 1,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 48,
                    "justifyContent": "center",
                    "width": 48,
                  },
                  Object {
                    "height": 48,
                    "width": 48,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "marginLeft": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "500",
                },
                false,
                undefined,
              ]
            }
          >
            Foo bar test center element
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              false,
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={undefined}
            testID={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderRadius": 21.6,
                  "height": 43.2,
                  "left": 2.3999999999999986,
                  "opacity": 0.16,
                  "position": "absolute",
                  "top": 2.3999999999999986,
                  "transform": Array [
                    Object {
                      "scale": 0.01,
                    },
                  ],
                  "width": 43.2,
                  "zIndex": 1,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 48,
                    "justifyContent": "center",
                    "width": 48,
                  },
                  Object {
                    "height": 48,
                    "width": 48,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Some dummy child component
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Another dummy child component
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
